Automattic/sensei
# Change/enrol users bulk management

#### Changes proposed in this Pull Request:
This PR updates the bulk management screens to use enrolment. 

There are 5 different actions that are offered. Only two of them have been functionaly changed, 'Assign to Course' and 'Unassign from Course. They have been changed to 'Manually enroll' and 'Remove manual enrollment' respectively. The rest remained the same.

Notes:
- It is advised to review the changes commit by commit. There has been a lot of cleanup to make the files to adhere to the coding standards so I have tried to separate the cleanup with the actual changes.
- There were a lot of cases where we use GET arguments without nonces. Since the arguments are not used to change data, I ignored the warnings. A double-check for these would be useful.
- I directly removed a few public attributes/methods. These were mainly unused code and/or code that didn't do anything (probably leftover from copied code) so I think that it is highly unlikely to break users. This is the list of what was removed:
```
Sensei_Learners_Admin_Bulk_Actions_Controller
- method get_query_args
- method learner_admin_menu
- const ADD_TO_COURSE
- const REMOVE_FROM_COURSE
- const RESET_COURSE

Sensei_Learners_Admin_Bulk_Actions_View
- public attribute view
- public attribute page_slug
- dynamic attributes $course_id, $lesson_id
```

#### Testing instructions:
All bulk learner actions should be tested. 'Manually enroll' and 'Remove manual enrollment' should give or take away manuall access to a learner. The rest should be the same.

Edit: Check also that the modal which asks users to join the mailing list appears. To make it appear use the following query `insert into wp_options (option_name,option_value) values ('sensei_show_email_signup_form',1);` and refresh.

Edit: The functionality of filtering by course has been changed. Now filtering does not affect the Overview column in any way. The column has been renamed to course progress. To test this, filter by course and observe the contents of the 'Course progress' column.
